drivers/i2c/i2c_ll_stm32.c

```diff
@@ -156,6 +157,9 @@ static int i2c_stm32_transfer(const struct device *dev, struct i2c_msg *msg,
 	pm_device_busy_set(dev);
 #endif
 
+	/* Prevent the clocks to be stopped during the i2c transaction */
+	pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
```

Comment:
@ceolin would you mind providing some guidance here ?

I guess the case here is when only CONFIG_PM is used and hence `pm_device_busy_set()` has no effect.

Should we really implement the following in such case ?
```
#ifdef CONFIG_PM_DEVICE_RUNTIME
	(void)pm_device_runtime_get(dev);
#elif CONFIG_PM_DEVICE
	pm_device_busy_set(dev);
#else
       pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
#endif
```

Reply:
I don't think that this will work because suspend to idle happens regardless of the device runtime state. We need to disable suspend to idle when power management is enabled.

Reply:
Ok, right. Point might then be that `pm_device_busy_set()` is actually mostly useless in STM32 drivers.

Reply:
@coran21 
In Idle function `pm_system_suspend()` will be used to trigger PM related actions.
In the algo, call to `pm_policy_next_state()` will end up in `pm_policy_state_lock_is_active`.
At this step, if `pm_policy_state_lock_get()` has been previously called system suspension is cancelled, `pm_suspend_devices()` doesn't happen and `pm_device_busy_set()` has no effect in any way.

So I agree, we should keep `pm_device_runtime_get()`, but `pm_device_busy_set()` becomes a no-op according to current code and should be removed.

Reply:
@coran21 Do you have plans to continue this PR ?

Reply:
Yes, I would like to finish this because I2C currently doesn't work at all with power management enabled on STM32L4 (or at least for me). I was waiting for the feedback from @ceolin. Should I continue with removing `pm_device_busy_set()` as you mentioned above?

Reply:
Maybe you can have a try and confirm this is working.
On my side, I would approuve such change

Reply:
I can confirm that this works right away. Patching the I2C driver with these changes is the first thing I have to do to get working EEPROM in my project after I update the Zephyr source code.

Reply:
@coran21 Sorry if this wasn't clear enough, but the request was to remove `pm_device_busy_set()` and test.

Reply:
I removed `pm_device_busy_set()` and tested it and it's exactly how you wrote - this function doesn't have any effect if `pm_policy_state_lock_get()` was called before.

Reply:
It looks like there were some changes in the actual file which resulted in conflict. Could you help me how I can update my local branch `fix_stm32_i2c_disable_suspend` with these changes so that I can push the changes again?  I know that I should do some kind of rebase, but it happened to me a few times that I did a rebase and force-pushed all the previous commits, which were rebased into the pull request, and it was a mess.
In Zephyr documentation is that I should do this while I have checked out my branch with changes:
```
git fetch --all
git rebase --ignore-whitespace upstream/main
```
And do interactive rebase of my commit, is that true? I'm really not sure what to do if there is a conflict in my commit.

Reply:
I initially thought that I can just resolve the conflicts in the Github editor and merge them, but that didn't work.

Reply:
@erwango, Could you help me with that? How could I update my local branch, which has conflicts with Zephyr:main, so that it won't push all the Zephyr commits that were merged after this commit after I do force-push to this PR branch?

Reply:
You are right, in this case `pm_device_busy_set` is useless if you have a lock in the power state that triggers device pm. The only thing you should keep in mind is that when a device is set busy, it won't be suspended in any power state transition. Here, the only when transitioning to `PM_STATE_SUSPEND_TO_IDLE` the device won't be suspended.

Reply:
@erwango @coran21  it looks reasonable to me. If you need to lock a power state to avoid other components (clocks) to be suspended as well, setting the busy state for only this is useless. The only thing that I am thinking is if there are other power states in the target that would also trigger device power management. If there isn't any other one, than this change is correct imho.